flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/source/KafkaSourceBuilder.java

```diff
@@ -438,9 +457,13 @@ private void parseAndSetRequiredProperties() {
             maybeOverride(KafkaSourceOptions.COMMIT_OFFSETS_ON_CHECKPOINT.key(), "false", false);
         }
         maybeOverride(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false", false);
+
         maybeOverride(
                 ConsumerConfig.AUTO_OFFSET_RESET_CONFIG,
-                startingOffsetsInitializer.getAutoOffsetResetStrategy().name().toLowerCase(),
+                startingOffsetsInitializer
```

Comment:
Could be NPE? You removed the initialization for `startingOffsetsInitializer` from the constructor.

UPDATE: I see, that `sanityCheck()` is called before `parseAndSetRequiredProperties()` - and this allows us to escape from NullPointerException. However, this approach seems very fragile to me. I suggest to set `startingOffsetsInitializer` in `parseAndSetRequiredProperties()`.